tests/ui/unnecessary_reserve.rs

```diff
@@ -0,0 +1,95 @@
+#![warn(clippy::unnecessary_reserve)]
+#![feature(custom_inner_attributes)]
+
+use std::collections::HashMap;
+use std::collections::VecDeque;
+
+fn main() {
+    vec_reserve();
+    vec_deque_reserve();
+    hash_map_reserve();
+    msrv_1_62();
+}
+
+fn vec_reserve() {
+    let mut vec: Vec<usize> = vec![];
+    let array: &[usize] = &[1, 2];
+
+    // do not lint
+    vec.reserve(1);
+    vec.extend([1]);
```

Comment:
I think one test that is missing is something like

```
vec.reserve(array1.len());
vec.extend(array2);
```

This might be covered by this, but I'm not 100% sure.

This should _not_ be linted. Doesn't make much sense to write something like this, but it should be an easy FP to avoid.

Reply:
Ok.